### PR TITLE
Autoconfigure Config

### PR DESCRIPTION
**Pull request by mackjmr**

### CODE REVIEW

File: comp/otelcol/converter/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- Taking in a user provided configuration, and enhancing the config
+- Providing an API which returns the provided and enhanced config
+
+## Autoconfigure logic
+
+The autoconfigure logic is applied within the `Convert` function. It takes in a `*confmap.Conf` and will modify it based on the logic described below.
+
+### Extensions
+
+The converter looks for the `pprof`, `health_check` and `zpages` extensions. If these are defined in the service pipeline, it does nothing. If any of these extensions are not defined, then it will add it's own extension config (name: `<extension_name>/dd-autoconfigured`) and add this in the services extension pipeline.
```
Comment (dineshg13) [inline comp/otelcol/converter/README.md:13]: It should add `datadog` extension

File: comp/otelcol/converter/impl/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+													}
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
```
Comment (dineshg13) [inline comp/otelcol/converter/impl/prometheus.go:97]: Avoid nested loops, by returning if `!ok` .  That is preferred golang style.

**Review by dineshg13 — CHANGES_REQUESTED** [review]

File: comp/otelcol/converter/README.md (hunk 1 above)
Comment (mackjmr) [reply comp/otelcol/converter/README.md]: Added `datadog` extension 👍

File: comp/otelcol/converter/impl/prometheus.go (hunk 2 above)
Comment (mackjmr) [reply comp/otelcol/converter/impl/prometheus.go]: Replaced by `return` or `continue`. I think this makes the code a bit easier to understand, but there are a lot of:
```
if !ok {
    return
}
```
```
if !ok {
    continue
}
```

**Review by dineshg13 — APPROVED** [review]
LGTM

File: comp/otelcol/converter/impl/prometheus.go (hunk 2 above)
Comment (dineshg13) [reply comp/otelcol/converter/impl/prometheus.go]: That is fine. and is preferred golang style.

File: comp/otelcol/converter/impl/infraattributes.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			return
+		}
+		exporters, ok := componentsMap["exporters"]
+		if !ok {
+			continue
+		}
+		exportersSlice, ok := exporters.([]any)
+		if !ok {
+			return
+		}
```
Comment (dineshg13) [inline comp/otelcol/converter/impl/infraattributes.go:57]: We are doing this in multiple places. We can combine them

**Pull request merged**